AlphaWallet/Settings/Coordinators/ExportJsonKeystoreCoordinator.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  ExportJsonKeystoreCoordinator.swift
+//  AlphaWallet
+//
+//  Created by Jerome Chan on 1/12/21.
+//
+
+import Foundation
+import UIKit
+
+@objc protocol ExportJsonKeystoreCoordinatorCallback {
+    func didComplete(coordinator: ExportJsonKeystoreCoordinator)
+}
+
+@objc protocol ExportJsonKeystoreInfoDelegate {
+    func didRequestPassword()
+    func didDismissInfoController()
+}
+
+@objc protocol ExportJsonKeystorePasswordDelegate {
+    func didRequestExportKeystore(with password: String)
+    func didDismissPasswordController()
+}
+
+class ExportJsonKeystoreCoordinator: NSObject, Coordinator {
+    enum StartMode {
+        case info
+        case password
+    }
+
+    var coordinators: [Coordinator] = []
+    weak var delegate: ExportJsonKeystoreCoordinatorCallback?
+    private var startMode: StartMode = .info
+    private var keystore: Keystore
+    private var password: String = ""
+    private var navigationController: UINavigationController
+    private weak var initialViewController: UIViewController?
+
+    init(keystore: Keystore, navigationController: UINavigationController) {
+        self.keystore = keystore
+        self.navigationController = navigationController
+        initialViewController = navigationController.topViewController
+    }
+
+    func start(startMode: StartMode) {
+        self.startMode = startMode
+        switch self.startMode {
+        case .info:
+            startInfoViewController(buttonTitle: R.string.localizable.settingsAdvancedExportJSONKeystoreInfoPasswordButtonInfo())
+        case .password:
+            startPasswordViewController(buttonTitle: R.string.localizable.settingsAdvancedExportJSONKeystorePasswordPasswordButtonPassword())
+        }
+    }
+
+    private func startInfoViewController(buttonTitle: String) {
+        let controller = ExportJsonKeystoreInfoViewController(viewModel: ExportJsonKeystoreInfoViewModel(keystore: keystore), buttonTitle: buttonTitle)
+        controller.infoDelegate = self
+        navigationController.pushViewController(controller, animated: true)
+    }
+
+    private func startPasswordViewController(buttonTitle: String) {
+        let validator: StringValidator = StringValidator(rules: [
```

Comment:
I think the definition of the validation rules should be in the view controller (or view model, or model), not in the coordinator.

Reply:
It should be in the model but we don't have any models. It's passed to the ViewModel. Do you want the ViewModel to automatically have these rules?

Reply:
Yes. Sounds a bit better than stuffing it in the coordinator (we already do too much of that in existing code)